WalletWasabi/Blockchain/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -273,6 +273,33 @@ private ProcessedResult ProcessNoLock(SmartTransaction tx)
 			TransactionStore.AddOrUpdate(tx);
 		}
 
+		if (tx.Confirmed)
+		{
+			// If a key is internal and spent all its coins, then it shouldn't be used again.
+			foreach (var spenderKey in myInputs.Select(x => x.HdPubKey).Where(x => x.IsInternal).Distinct())
```

Comment:
Comment: "Update for Turbosync - mark internal keys as spent if there is a spender tx. "